cogs/overwatch2.py

```diff
@@ -71,57 +167,231 @@ def __init__(self, client):
             "Grandmaster": "<:Grandmaster:1109604769963716688>",
             "Top 500": "<:Top500:1109604938297905293>",
         }
-        with open("players.json", 'r') as f:
-            self.players = json.load(f)
-        '''with open("ranks.json", 'r') as f:
-            self.ranks = json.load(f)'''
-
-    @commands.has_permissions(administrator=True)
-    @commands.command()
-    async def overwatch(self, ctx):
-        msg = self._alg()
-        await ctx.send(msg)
-
-    def _alg(self) -> str:
-        p = list(self.players.keys())
-        p.remove("Mitch")
-        while True:
+        self.players = PLAYERS
+
+    def _get_emoji(self, name, role) -> str:
+        return self.role_emojis[self.players[name][role.lower()][:-2]]
+
+    async def role_queue(self, interaction: discord.Interaction, timeout: int):
+        view = discord.ui.View()
+        select = RoleQueueSelect(interaction)
+        view.add_item(select)
+        await interaction.response.send_message("Choose role queues", view=view)
+        i = timeout
+        msg = await interaction.followup.send(content=".")
+        while i >= 0:
+            await interaction.followup.edit_message(
+                msg.id, content=f"**({i} seconds remaining)**")
+            await asyncio.sleep(1)
+            i -= 1
+        interaction.is_expired()
+        await asyncio.sleep(2)
+        await interaction.followup.delete_message(msg.id)
+        await interaction.delete_original_response()
+        return select.queues
+
+    @app_commands.checks.has_permissions(moderate_members=True)
+    @app_commands.command(
+        name="overwatch5v5",
+        description="Creates a 5v5 matchup for Overwatch"
+    )
+    @app_commands.describe(timeout="Time to select role", flex="For testing") # to become for flexing
+    async def overwatch(self, interaction: discord.Interaction,
+                        timeout: app_commands.Range[int, 5, 120] = 15,
+                        flex: Bool = Bool.false) -> None:
+
+        if self.active:
+            await interaction.response.send_message(
+                "A game is currently active. This command has no effect",
+                ephemeral=True
+            )
+            return
+
+        p = list(PLAYERS.keys())
+        if flex == Bool.true: # for testing cuz i dont wanna make proper test :D
+            await interaction.response.defer()
+            queues: dict[str, list[str]] = {player: ['Damage', 'Support'] for player in p}
+            queues["ponky"] = ['Tank', 'Damage', 'Support']
+            queues["trk20"] = ['Tank', 'Damage', 'Support']
+        else:
+            queues = await self.role_queue(interaction, timeout)
+
+        if len(queues) < 10:
+            await interaction.followup.send(f"Not enough players queued: **{len(queues)} queued**")
+            return
+
+        bad_teams = 0
+        while bad_teams != 1000:
```

Comment:
Might be a good idea to change later to a method that uses itertools.combinations or similar then filter result - current one can generate the same thing twice

Reply:
Something like this
```
combinations.([array of players], 5) //252 if 10 players
product(combos,2) // 63K
[[x, y] for x in combos for y in combos if not set(x).intersection(set(y))] //filter out duplicate players
//search through result
```
Not sure if it would need to be multithreaded/async but this method would mean it only needs to be done once per rank update instead of once per match request